src/MIDebugEngine/AD7.Impl/AD7Expression.cs

```diff
@@ -88,5 +89,33 @@ int IDebugExpression2.EvaluateSync(enum_EVALFLAGS dwFlags, uint dwTimeout, IDebu
         }
 
         #endregion
+
+        #region IDebugExpressionDAP
+
+        int IDebugExpressionDAP.EvaluateSync(enum_EVALFLAGS dwFlags, DAPEvalFlags dapFlags, uint dwTimeout, IDebugEventCallback2 pExprCallback, out IDebugProperty2 ppResult)
+        {
+            ppResult = null;
+            if ((dwFlags & enum_EVALFLAGS.EVAL_NOSIDEEFFECTS) != 0 && _var.IsVisualized)
+            {
+                IVariableInformation variable = _engine.DebuggedProcess.Natvis.Cache.Lookup(_var);
+                if (variable == null)
+                {
+                    ppResult = new AD7ErrorProperty(_var.Name, ResourceStrings.NoSideEffectsVisualizerMessage);
+                }
+                else
+                {
+                    _var = variable;
+                    ppResult = new AD7Property(_engine, _var);
+                }
+                return Constants.S_OK;
+            }
+
+            _var.SyncEval(dapFlags, dwFlags);
```

Comment:
Could you have the top method call this method with dapFlags.None and then fix the order of your flags to have SyncEval use dwFlags and then dapFlags? i think the switching of order of which the flags are in different methods is going to cause confusion.